ingestion/src/metadata/ingestion/source/sql_source.py

```diff
@@ -102,6 +104,114 @@ def _get_table_description(schema: str, table: str, inspector: Inspector) -> str
     return description
 
 
+def get_dbt_local(config) -> Optional[Tuple[str, str]]:
+    try:
+        if config.dbtConfig.dbtCatalogFilePath is not None:
+            with open(
+                config.dbtConfig.dbtCatalogFilePath, "r", encoding="utf-8"
+            ) as catalog:
+                dbt_catalog = catalog.read()
+        if config.dbtConfig.dbtManifestFilePath is not None:
+            with open(
+                config.dbtConfig.dbtManifestFilePath, "r", encoding="utf-8"
+            ) as manifest:
+                dbt_manifest = manifest.read()
+        return json.loads(dbt_catalog), json.loads(dbt_manifest)
+    except Exception as exc:
+        logger.debug(traceback.format_exc())
+        logger.debug(f"Error fetching dbt files from local {repr(exc)}")
+    return None
+
+
+def get_dbt_http(config) -> Optional[Tuple[str, str]]:
+    try:
+        catalog_file = urllib.request.urlopen(config.dbtConfig.dbtCatalogFilePath)
+        manifest_file = urllib.request.urlopen(config.dbtConfig.dbtManifestFilePath)
+        dbt_catalog = catalog_file.read().decode()
+        dbt_manifest = manifest_file.read().decode()
+        return json.loads(dbt_catalog), json.loads(dbt_manifest)
+    except Exception as exc:
+        logger.debug(traceback.format_exc())
+        logger.debug(f"Error fetching dbt files from http {repr(exc)}")
+    return None
+
+
+def get_dbt_gcs(config) -> Optional[Tuple[str, str]]:
+    try:
+        dbt_options = config.dbtConfig.gcsConfig
```

Comment:
@pmbrull the GCSCredentials model is used here to fetch gcs credentials and then it is converted to the required format for gcs